pkg/controllers/management/restrictedadminrbac/fleet_test.go

```diff
@@ -0,0 +1,118 @@
+package restrictedadminrbac
+
+import (
+	"testing"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/rbac"
+	k8srbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_rbaccontroller_ensureRolebinding(t *testing.T) {
```

Comment:
I'd also recommend adding tests for the following cases: 
- We get an error other than not found on the get call, that gets returned up
- We get an error when creating the binding, that gets returned up
- The existing role has different ownerrefs, we update
- The existing role has different rolerefs, we update
- The existing role has different subjects, we update
- We tried updating but got an error, that is returned up

Lastly, generally speaking it's a good practice to test the handler itself (i.e. the `ensureRestricedAdminForFleet` function) since those are closer to "public" functions that have an input/output interface (rather than private functions that serve as implementation details). In this case, the public function is relatively simple, and we could get some additional coverage out of testing that directly. But since you didn't actually change it, I don't think it's required to get this PR in.

Reply:
Suggested tests added as of 7fc80114d